src/CoreWCF.Primitives/src/CoreWCF/Description/MessagePartDescription.cs

```diff
@@ -48,12 +48,12 @@ internal MessagePartDescription(MessagePartDescription other)
             //this.uniquePartName = other.uniquePartName;
         }
 
-        internal virtual MessagePartDescription Clone()
+        public virtual MessagePartDescription Clone()
         {
             return new MessagePartDescription(this);
         }
 
-        internal XmlName XmlName { get; }
+        public XmlName XmlName { get; }
 
```

Comment:
This doesn't need to be made public. XmlName is constructed with an already encoded name, so the Name property will return the original encoded string passed to the constructor. This property is used in UriTemplateClientFormatter which just calls XmlName.DecodedName. If you look at the implementation of DecodedName, it all comes down to `XmlConvert.DecodeName` which is public. You can leave this property internal, leave XmlName internal, and modify UriTemplateClientFormatter  to use the code:
```c#
string parameterName = XmlConvert.DecodeName(mpd.Name)
```;

Reply:
This is addressed in 75f6cfa72e482b0f0f42ad137be5f9b694044554.